### PR TITLE
add support for timeout feature

### PR DESCRIPTION
**Pull request by kristopherbullinger**
Hello. Discord released a new feature recently allowing moderators/admins to time out users for a specified amount of time. Timing out a user requires a [new permission](https://discord.com/developers/docs/topics/permissions#permissions) and is part of the [Modify Guild Member](https://discord.com/developers/docs/resources/guild#modify-guild-member) API.

Thank you for your time reading and reviewing this pull request. I am happy to address any issues if you find that there are any.

### CODE REVIEW

File: src/client/mod.rs [hunk 1]
```diff
@@ -369,7 +369,6 @@ impl<'a> Future for ClientBuilder<'a> {
 
             self.fut = Some(Box::pin(async move {
                 let url = Arc::new(Mutex::new(http.get_gateway().await?.url));
-
```
Comment (nickelc) [inline src/client/mod.rs]: This change is unrelated.

File: src/builder/edit_member.rs [hunk 2]
```diff
@@ -87,4 +90,40 @@ impl EditMember {
...
+        self.0.insert("communication_disabled_until", Value::String(timestamp));
+        self
+    }
+
+    /// Remove a user's timeout, if there is one.
+    ///
+    /// Requires the [Moderate Members] permission.
+    ///
+    /// [Moderate Members]: crate::model::permission::Permissions::MODERATE_MEMBERS
+    pub fn remove_timeout(&mut self) -> &mut Self {
```
Comment (GnomedDev) [inline src/builder/edit_member.rs]: Seems like a weird name, maybe the other methods should take Option to follow the API or this should be renamed to `enable_communication`

**Review by GnomedDev — COMMENTED** [review]
This PR also doesn't seem to add the new `communication_disabled_until` field to Member, however it may be missing something.

File: src/client/mod.rs (hunk 1 above)
Comment (kristopherbullinger) [reply src/client/mod.rs]: I will undo this change. My mistake, left over from some debugging during local testing.

File: src/builder/edit_member.rs (hunk 2 above)
Comment (kristopherbullinger) [reply src/builder/edit_member.rs]: I will update this method name.

**Comment by kristopherbullinger** [comment]
In total, I have:
- Updated `Member` and `PartialMember` to contain a new `communication_disabled_until: Option<DateTime<Utc>>` field.
- Added `disable_communication_until_datetime` and `enable_communication` methods to the `Member` struct.
- Added three new methods to `EditMember`: `disable_communication_until`, `disable_communication_until_datetime`, and `enable_communication`.

I am wondering if it is better for the `EditMember::disable_communication_until_datetime` to accept as a parameter `DateTime<Utc>` rather than `DateTime<Tz> where Tz: TimeZone, Tz::Offset: Display` to simplify the API.

File: src/model/guild/member.rs [hunk 3]
```diff
@@ -92,7 +97,7 @@ impl Member {
                 self.roles.push(role_id);
 
                 Ok(())
-            },
+            }
```
Comment (kristopherbullinger) [inline src/model/guild/member.rs]: These changes kept getting applied every time I ran RustFmt. Let me know if you want me to undo them.

Comment (arqunis) [reply src/model/guild/member.rs]: These commas are meant to be there, as [defined by our rustfmt configuration](https://github.com/serenity-rs/serenity/blob/current/rustfmt.toml#L11). Which version of rustfmt are you using? We only support nightly, because we use options not yet supported in stable rustfmt.

Comment (kristopherbullinger) [reply src/model/guild/member.rs]: I see. I was not using nightly -- I was letting an editor plugin do the formatting for me, which was using the default rustfmt channel. Running `cargo +nightly fmt` resolved the commas and other formatting discrepancies. Thank you.

**Review by arqunis — APPROVED** [review]

**Comment by arqunis** [comment]
Thanks!
